src/main/java/software/amazon/awssdk/sar/ClientBuilderUsingRetryStrategy.java

```diff
@@ -0,0 +1,154 @@
+import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.core.retry.RetryUtils;
+import software.amazon.awssdk.retries.AdaptiveRetryStrategy;
+import software.amazon.awssdk.retries.api.BackoffStrategy;
+import software.amazon.awssdk.retries.api.RetryStrategy;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+/**
+ * Building clients using {@link RetryStrategy} is similar to using a {@link RetryPolicy}.
+ */
+public class ClientBuilderUsingRetryStrategy {
+    /**
+     * Using retry mode is handy and supported by
+     * {@link ClientOverrideConfiguration.Builder#retryStrategy(RetryMode)}
+     */
+    public static DynamoDbClient clientWithRetryStrategy01() {
+        return DynamoDbClient
+                .builder()
+                .overrideConfiguration(o -> o.retryStrategy(RetryMode.STANDARD))
+                .build();
+    }
+
+    /**
+     * The new {@link RetryMode#ADAPTIVE} is <em>also</em> supported by retry strategies, but we don't have a concrete
+     * implementation of it, instead behind the scenes it uses the policy to strategy internal adapter
+     * {@link RetryPolicyAdapter} built to support backwards compatibility but not expected to be used by customers (it
+     * is an internal API).
+     */
+    public static DynamoDbClient clientWithRetryStrategy03() {
+        // Does *not* throws UnsupportedOperationException.
+        return DynamoDbClient
+                .builder()
+                .overrideConfiguration(o -> o.retryStrategy(RetryMode.ADAPTIVE))
+                .build();
+    }
+
+    /**
+     * RetryStrategy is an interface parametrized on its concrete implementation and its builder, the full signature of
+     * the type is
+     * {@snippet
+     * public interface RetryStrategy<B extends CopyableBuilder<B, T> & RetryStrategy.Builder<B, T>,
+     *                                T extends ToCopyableBuilder<B, T> & RetryStrategy<B, T>>
+     *                      extends ToCopyableBuilder<B, T> { ⋯
+     * }
+     * }
+     * <p>
+     * This causes that when we don't use a concrete type of it we need to refer to it as {@code RetryStrategy<?, ?>}
+     * 😔
+     *
+     * @see RetryStrategy
+     * @see RetryStrategy.Builder
+     */
+    public static DynamoDbClient clientWithRetryStrategy04() {
+        RetryStrategy<?, ?> strategy = AwsRetryStrategy.forRetryMode(RetryMode.ADAPTIVE_V2)
+                // toBuilder RetryStrategy.Builder which can configure some basic properties
+                .toBuilder()
+                // Legacy numRetries(N) equivalent is
+                // maxAttempts(N + 1) (includes the first attempt)
+                .maxAttempts(121)
+                // Legacy BackoffStrategy.none() equivalent is
+                // BackoffStrategy.retryImmediately()
+                .backoffStrategy(BackoffStrategy.retryImmediately())
```

Comment:
How does retryImmediately work with adaptive?

Reply:
The backoff strategy duration is just used, the way it's currently implemented is that the resulting duration is `rate-limiter-duration + backoff-duration`. See [here](https://github.com/aws/aws-sdk-java-v2/blob/feature/master/sra-retries/core/retries/src/main/java/software/amazon/awssdk/retries/internal/DefaultAdaptiveRetryStrategy.java#L56)